gobblin-runtime/src/main/java/gobblin/runtime/listeners/JobListeners.java

```diff
@@ -172,6 +173,25 @@ public void close() throws IOException {
         }
       } finally {
         ExecutorsUtils.shutdownExecutorService(this.executor, Optional.of(LOGGER));
+        closeJobListeners();
+      }
+    }
+
+    private void closeJobListeners() throws IOException {
+      IOException exception = null;
+      for (JobListener jobListener : this.jobListeners) {
+        if (jobListener instanceof Closeable) {
+          try {
+            ((Closeable) jobListener).close();
+          } catch (IOException e) {
+            if (exception == null) {
```

Comment:
I'll add a log message, but this code will store the first caught exception and re-throw it once all the listeners are closed. This is the same behavior as not catching the exception, but ensures all listeners are closed.